### PR TITLE
[Migrator] Update spec checksums in the manifest

### LINKED ISSUE #3292: `pod update`ing a specific messes up other pods referenced by tag or commit (0.36)
This only happens on `0.36` (works fine on `0.35`).

I tried to `pod update some-specific-pod`, then that [incorrectly] updated the references for all other dependencies that are referenced by git repo + (commit|tag). It seems to have made them point to really old versions. Let me know what other info I can give!

:octocat: 

P.S. Congrats on the Swift release! :confetti_ball:

### LINKED ISSUE #3299: :parameters in Podfile not taken into account in 0.36
Hi,
I had to downgrade to 0.35 because on 0.36 running `pod install` doesn't take into account the `:git`, `commit` and `:branch` parameter being passed.

These are some entries of my `Podfile` that are being installed but without respecting the extra parameters.

```
pod 'SocketRocket',               :git => 'https://github.com/FredericJacobs/SocketRocket.git', :commit => '73d8a19'
pod 'JSQMessagesViewController',  :git => 'https://github.com/WhisperSystems/JSQMessagesViewController', :branch => 'JSignalQ'
pod 'APDropDownNavToolbar',       :git => 'https://github.com/corbett/APDropDownNavToolbar.git', :branch => 'master'
pod 'EZAudio',                    :git => 'https://github.com/jackflips/EZAudio', :commit => '1620f19'
```

(Quickly looked through issues but couldn't find another instance of this)

### PR DESCRIPTION
**Pull request by segiddins**
Closes #3299.
Probably also closes #3292.

\c @kylef @ayanonagon

### CODE REVIEW

File: lib/cocoapods/installer/migrator.rb [hunk 1]
```diff
@@ -53,13 +55,16 @@ def migrate_to_0_36(sandbox)
...
+            lockfile = sandbox.manifest.to_hash
             sandbox.specifications_root.children.each do |child|
               next unless child.basename.to_s =~ /\.podspec$/
               spec = Specification.from_file(child)
               child.delete
               child = Pathname("#{child}.json")
               File.open(child, 'w') { |f| f.write spec.to_pretty_json }
+              lockfile['SPEC CHECKSUMS'][spec.name] = Specification.from_file(child).checksum
             end
+            File.open(sandbox.manifest_path, 'w') { |f| f.write Lockfile.new(lockfile).to_yaml }
```
Comment (kylef) [inline lib/cocoapods/installer/migrator.rb]:
```ruby
sandbox.manifest_path.write(Lockfile.new(lockfile).to_yaml)
```

?

Comment (segiddins) [inline lib/cocoapods/installer/migrator.rb]: `Pathname#write` isn't available on 2.0.0?

Comment (kylef) [inline lib/cocoapods/installer/migrator.rb]: Keep forgetting that :'(

**Comment by ayanonagon** [comment]
Yayyy. That was super fast, thank you! :octocat:

**Comment by segiddins** [comment]
@ayanonagon can you confirm that this fixes things?

File: lib/cocoapods/installer/migrator.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@ class Installer
     #
     class Migrator
       class << self
+        extend Config::Mixin
```
Comment (segiddins) [inline lib/cocoapods/installer/migrator.rb]: this is unnecessary

**Comment by ayanonagon** [comment]
@segiddins Sure! How do I do that? i.e. How do I run that version of CocoaPods locally?

**Comment by kylef** [comment]
I think the easiest way would be:

```
$ gem install specific_install
$ gem specific_install -l https://github.com/CocoaPods/CocoaPods -b seg-0.36-migration
```

**Comment by segiddins** [comment]
http://guides.cocoapods.org/using/unreleased-features

**Pull request merged**